### PR TITLE
Vsitnik new attribs

### PR DESCRIPTION
**Pull request by vsitnik**
Enabling "species.annotation_source" and "genebuild.method_display" meta data (optional keys) support.

Loading pipeline loads.

### CODE REVIEW

File: scripts/gff_metaparser/gen_meta_conf.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def get_args():
                       type=str, default = "EnsemblMetazoa", help="species.division default")
   parser.add_argument("--genebuild_method", metavar="import", required = False,
                       type=str, default = "import", help="genebuild.method default")
+  parser.add_argument("--genebuild_method_display", metavar="Import", required = False,
+                      type=str, default = "Import", help="genebuild.method_display default")
```
Comment (JAlvarezJarreta) [inline scripts/gff_metaparser/gen_meta_conf.py:67]:
```suggestion
  parser.add_argument("--genebuild_method_display", metavar="Import", required=False,
                      type=str, default="Import", help="genebuild.method_display default")
```
To make it PEP8 compliant :upside_down_face:

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 2]
```diff
@@ -94,6 +94,16 @@ def update(self, key, val, tech = False):
     else:
       out[key].append(val)
 
+  def normalise_asm_name(self, asm_name):
+    if not asm_name:
+      return asm_name
```
Comment (JAlvarezJarreta) [inline scripts/gff_metaparser/genmetaconf/metaconf.py:99]: If `asm_name` can be `None`, then this is fine, but if not, you can just run the code below (will not do much and the performance impact will be minimal).

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 3]
```diff
@@ -159,13 +182,28 @@ def update_derived_data(self, defaults = None, update_annotation_related = False
...
+    #"_species.annotation_source_sfx" : args.annotation_source_sfx,
     self.update("species.production_name", _prod_name)
     _comm_name = self.get("species.common_name")
     _display_name = _sci_name
     if _strain or _comm_name:
       _strain_comm_part = ", ".join(map(lambda s: str(s), filter(None, [_comm_name, _strain])))
       _display_name += " (%s)" % _strain_comm_part
     _display_name += " - " + asm_acc
+    # possibly add annotation source tag
+    _ann_source = self.get("species.annotation_source", default = "").strip()
```
Comment (JAlvarezJarreta) [inline scripts/gff_metaparser/genmetaconf/metaconf.py:202]:
```suggestion
    _ann_source = self.get("species.annotation_source", default="").strip()
```

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 4]
```diff
@@ -159,13 +182,28 @@ def update_derived_data(self, defaults = None, update_annotation_related = False
...
+    _ann_source = self.normalise_asm_name(_ann_source)
+    if _ann_source:
+      self.update("species.annotation_source", _ann_source)
+      _display_name = f"%s [%s annotation]" % (_display_name, _ann_source)
```
Comment (JAlvarezJarreta) [inline scripts/gff_metaparser/genmetaconf/metaconf.py:206]:
```suggestion
      _display_name = f"{_display_name} [{_ann_source} annotation]"
```

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 5]
```diff
@@ -159,13 +182,28 @@ def update_derived_data(self, defaults = None, update_annotation_related = False
     _prod_name = _sci_name.strip().lower()
     _prod_name = "_".join(re.sub(r'[^a-z0-9A-Z]+', '_', _prod_name).split("_")[:2])
     _prod_name = ("%s_%s" % (_prod_name, _acc)).lower().replace(" ","_")
+    # possibly add annotation source suffix
+    _ann_source_sfx = self.get("ANNOTATION_SOURCE_SFX", tech = True, default = "").strip()
```
Comment (JAlvarezJarreta) [inline scripts/gff_metaparser/genmetaconf/metaconf.py:186]:
```suggestion
    _ann_source_sfx = self.get("ANNOTATION_SOURCE_SFX", tech=True, default="").strip()
```

**Review by JAlvarezJarreta — CHANGES_REQUESTED** [review]

File: scripts/gff_metaparser/genmetaconf/metaconf.py (hunk 2 above)
Comment (vsitnik) [reply scripts/gff_metaparser/genmetaconf/metaconf.py:99]: as expected

File: scripts/gff_metaparser/genmetaconf/metaconf.py (hunk 4 above)
Comment (vsitnik) [reply scripts/gff_metaparser/genmetaconf/metaconf.py:206]: makes sense

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 6]
```diff
@@ -159,13 +182,28 @@ def update_derived_data(self, defaults = None, update_annotation_related = False
...
+    _ann_source_sfx = self.normalise_asm_name(_ann_source_sfx).replace("_","").lower()[:2]
+    if _ann_source_sfx:
+      self.update("ANNOTATION_SOURCE_SFX", _ann_source_sfx, tech = True)
+      _prod_name += _ann_source_sfx
+    # defaults
+    # "species.annotation_source" : args.annotation_source_name,
+    #"_species.annotation_source_sfx" : args.annotation_source_sfx,
```
[the remark this answers is not in the document]
Comment (vsitnik) [reply scripts/gff_metaparser/genmetaconf/metaconf.py]: removed. thanks for spotting

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 7]
```diff
@@ -125,13 +135,26 @@ def merge_from_gbff(self, gbff_file):
...
+    _open = asm_rep_file.endswith(".gz") and gzip.open or open
+    with _open(asm_rep_file, 'rt') as asm_rep:
+      for line in asm_rep:
+        # # Assembly name:  cgigas_uk_roslin_v1
+        if re.match(r'#\s+Assembly name:', line):
+          (_tag, asm_name, *_rest) = line.split(sep = ":", maxsplit=1)
+          asm_name = self.normalise_asm_name(asm_name)
+          self.update("assembly.name", asm_name)
+          break
+    return
```
[the remark this answers is not in the document]
Comment (vsitnik) [reply scripts/gff_metaparser/genmetaconf/metaconf.py:156]: no. thanks.
I don't like "unanchored" idents

[the remark this answers is not in the document]
Comment (JAlvarezJarreta) [reply scripts/gff_metaparser/genmetaconf/metaconf.py:156]: No worries, updated comment with the relevant changes (following previous suggestions).

**Pull request merged**